source/source_integration_test.go

```diff
@@ -0,0 +1,546 @@
+		ClientEmail: src[models.ConfigClientEmail],
+		ProjectID:   src[models.ConfigProjectID],
+	}.Marshal()
+}
+
+func createTopic(ctx context.Context, projectID, topicID string, credential []byte) error {
+	client, err := pubsub.NewClient(ctx, projectID, option.WithCredentialsJSON(credential))
+	if err != nil {
+		return fmt.Errorf("new client: %w", err)
+	}
+	defer client.Close()
+
+	topic, err := client.CreateTopic(ctx, topicID)
+	if err != nil {
+		return fmt.Errorf("create topic: %w", err)
+	}
+	topic.Stop()
+
+	return nil
+}
+
+func createSubscription(ctx context.Context, projectID, topicID, subscriptionID string, credential []byte) error {
+	client, err := pubsub.NewClient(ctx, projectID, option.WithCredentialsJSON(credential))
+	if err != nil {
+		return fmt.Errorf("new client: %w", err)
+	}
+	defer client.Close()
+
+	topic := client.Topic(topicID)
+	defer topic.Stop()
+
+	if _, err = client.CreateSubscription(ctx, subscriptionID, pubsub.SubscriptionConfig{
+		Topic: topic,
+	}); err != nil {
+		return fmt.Errorf("create subscription: %w", err)
+	}
+
+	return nil
+}
+
+func cleanup(ctx context.Context, projectID, topicID, subscriptionID string, credential []byte) error {
+	client, err := pubsub.NewClient(ctx, projectID, option.WithCredentialsJSON(credential))
+	if err != nil {
+		return fmt.Errorf("new client: %w", err)
+	}
+	defer client.Close()
+
+	if err = client.Subscription(subscriptionID).Delete(ctx); err != nil {
+		return fmt.Errorf("delete subscription: %w", err)
+	}
+
+	if err = client.Topic(topicID).Delete(ctx); err != nil {
+		return fmt.Errorf("delete topic: %w", err)
+	}
+
+	return nil
+}
+
+func prepareData(messagesCount int, dstCfg map[string]string) (map[string]struct{}, error) {
+	const dataFmt = "{\"id\": %s}"
+
+	dest := destination.New()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	err := dest.Configure(ctx, dstCfg)
+	if err != nil {
+		return nil, fmt.Errorf("configure: %s", err.Error())
+	}
+
+	err = dest.Open(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("open: %s", err.Error())
+	}
+
+	prepared := make(map[string]struct{}, messagesCount)
+
+	for i := 0; i < messagesCount; i++ {
+		data := fmt.Sprintf(dataFmt, strconv.Itoa(i))
+
+		err = dest.WriteAsync(ctx, sdk.Record{
+			Payload: sdk.RawData(data),
+		}, func(ackErr error) error {
+			if ackErr != nil {
+				return fmt.Errorf("ack func: %s", ackErr.Error())
+			}
+
+			return nil
+		})
+		if err != nil {
+			return nil, fmt.Errorf("write async: %s", err.Error())
+		}
+
+		prepared[data] = struct{}{}
+	}
+
+	err = dest.Flush(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("flush: %s", err.Error())
+	}
+
+	cancel()
+
+	err = dest.Teardown(context.Background())
+	if err != nil {
+		return nil, fmt.Errorf("teardown: %s", err.Error())
+	}
+
+	return prepared, nil
+}
+
+func compare(records []sdk.Record, prepared map[string]struct{}) error {
+	for i := range records {
+		payload := string(records[i].Payload.Bytes())
```

Comment:
Is it possible to verify the other parts of a record (key, position etc.)

Reply:
I've added the `metadata` field for verification.

The `position` field is a randomly generated `uuid`, and the `key` field is empty. 

[Commit d54530e](https://github.com/conduitio-labs/conduit-connector-gcp-pubsub/commit/d54530e6262c4da5f0a178a4eb9f7007215be78f)